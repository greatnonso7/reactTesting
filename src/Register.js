import React from 'react';
import {
  View,
  Text,
  SafeAreaView,
  StyleSheet,
  TextInput,
  TouchableOpacity,
} from 'react-native';

class Register extends React.PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      first: '',
      last: '',
      username: '',
      email: '',
      password: '',
    };
  }

  handleEmailTextInput(text) {
    this.setState({email: text});
  }
  handleFirstNameTextInput(text) {
    this.setState({first: text});
  }
  handleLastNameTextInput(text) {
    this.setState({last: text});
  }
  handleUsernameTextInput(text) {
    this.setState({username: text});
  }
  handlePasswordTextInput(text) {
    this.setState({password: text});
  }
  render() {
    return (
      <SafeAreaView style={styles.container}>
        <View style={styles.body}>
          <TouchableOpacity onPress={() => this.props.navigation.goBack()}>
            <Text>{'<'}</Text>
          </TouchableOpacity>

          <Text style={styles.heading}>Register Screen</Text>
          <View style={styles.formContainer}>
            <View style={styles.formControl}>
              <TextInput
                textId="firstNameTextInput"
                style={styles.textInput}
                onChangeText={e => this.handleFirstNameTextInput(e)}
                placeholder="First Name"
              />
            </View>
            <View style={styles.formControl}>
              <TextInput
                textId="lastNameTextInput"
                style={styles.textInput}
                onChangeText={e => this.handleLastNameTextInput(e)}
                placeholder="Last Name"
              />
            </View>
            <View style={styles.formControl}>
              <TextInput
                textId="emailTextInput"
                style={styles.textInput}
                onChangeText={e => this.handleEmailTextInput(e)}
                placeholder="Email Name"
              />
            </View>
            <View style={styles.formControl}>
              <TextInput
                textId="userNameTextInput"
                style={styles.textInput}
                onChangeText={e => this.handleUsernameTextInput(e)}
                placeholder="Username"
              />
            </View>
            <View style={styles.formControl}>
              <TextInput
                textId="passwordTextInput"
                style={styles.textInput}
                onChangeText={e => this.handlePasswordTextInput(e)}
                placeholder="Password"
              />
            </View>
          </View>
          <TouchableOpacity
            textId="loginButton"
            activeOpacity={0.6}
            style={styles.buttonContainer}>
            <Text style={styles.buttonText}>Register</Text>
          </TouchableOpacity>
        </View>
      </SafeAreaView>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  body: {
    marginHorizontal: 20,
    marginTop: 20,
  },
  heading: {
    fontSize: 16,
    textAlign: 'center',
  },
  formContainer: {
    marginTop: 20,
  },
  formControl: {
    width: 350,
    height: 50,
    marginBottom: 10,
    borderWidth: 1,
    borderRadius: 10,
  },
  textInput: {
    padding: 15,
    fontSize: 20,
    height: 50,
  },
  buttonContainer: {
    marginTop: 50,
    height: 50,
    width: 350,
    backgroundColor: 'red',
    justifyContent: 'center',
    borderRadius: 10,
    alignItems: 'center',
  },
  buttonText: {
    fontSize: 20,
    color: 'white',
  },
});

export default Register;
