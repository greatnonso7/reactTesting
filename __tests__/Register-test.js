import React from 'react';
import renderer from 'react-test-renderer';
import Register from '../src/Register';

test('Register snapshot ', () => {
  const snap = renderer.create(<Register />).toJSON();

  expect(snap).toMatchSnapshot();
});

test('Should change the state if email is entered', () => {
  const instanceOf = renderer.create(<Register />).getInstance();
  instanceOf.handleEmailTextInput('jon@jon.com');
  expect(instanceOf.state.email).toEqual('jon@jon.com');
});

test('Should change the state if first name is entered', () => {
  const instanceOf = renderer.create(<Register />).getInstance();
  instanceOf.handleFirstNameTextInput('Chinonso');
  expect(instanceOf.state.first).toEqual('Chinonso');
});

test('Should change the state if last name is entered', () => {
  const instanceOf = renderer.create(<Register />).getInstance();
  instanceOf.handleLastNameTextInput('Ichoku');
  expect(instanceOf.state.last).toEqual('Ichoku');
});

test('Should change the state if username is entered', () => {
  const instanceOf = renderer.create(<Register />).getInstance();
  instanceOf.handleUsernameTextInput('chinonso7');
  expect(instanceOf.state.username).toEqual('chinonso7');
});

test('Should change the state if password is entered', () => {
  const instanceOf = renderer.create(<Register />).getInstance();
  instanceOf.handlePasswordTextInput('12345678');
  expect(instanceOf.state.password).toEqual('12345678');
});
