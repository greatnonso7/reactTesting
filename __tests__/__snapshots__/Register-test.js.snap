// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register snapshot  1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "marginHorizontal": 20,
        "marginTop": 20,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text>
        &lt;
      </Text>
    </View>
    <Text
      style={
        Object {
          "fontSize": 16,
          "textAlign": "center",
        }
      }
    >
      Register Screen
    </Text>
    <View
      style={
        Object {
          "marginTop": 20,
        }
      }
    >
      <View
        style={
          Object {
            "borderRadius": 10,
            "borderWidth": 1,
            "height": 50,
            "marginBottom": 10,
            "width": 350,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="First Name"
          rejectResponderTermination={true}
          style={
            Object {
              "fontSize": 20,
              "height": 50,
              "padding": 15,
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        style={
          Object {
            "borderRadius": 10,
            "borderWidth": 1,
            "height": 50,
            "marginBottom": 10,
            "width": 350,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Last Name"
          rejectResponderTermination={true}
          style={
            Object {
              "fontSize": 20,
              "height": 50,
              "padding": 15,
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        style={
          Object {
            "borderRadius": 10,
            "borderWidth": 1,
            "height": 50,
            "marginBottom": 10,
            "width": 350,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Email Name"
          rejectResponderTermination={true}
          style={
            Object {
              "fontSize": 20,
              "height": 50,
              "padding": 15,
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        style={
          Object {
            "borderRadius": 10,
            "borderWidth": 1,
            "height": 50,
            "marginBottom": 10,
            "width": 350,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Username"
          rejectResponderTermination={true}
          style={
            Object {
              "fontSize": 20,
              "height": 50,
              "padding": 15,
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        style={
          Object {
            "borderRadius": 10,
            "borderWidth": 1,
            "height": 50,
            "marginBottom": 10,
            "width": 350,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Password"
          rejectResponderTermination={true}
          style={
            Object {
              "fontSize": 20,
              "height": 50,
              "padding": 15,
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "red",
          "borderRadius": 10,
          "height": 50,
          "justifyContent": "center",
          "marginTop": 50,
          "opacity": 1,
          "width": 350,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
          }
        }
      >
        Register
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
